src/seedu/addressbook/data/tag/Tagging.java

```diff
@@ -0,0 +1,71 @@
+package seedu.addressbook.data.tag;
+
+import seedu.addressbook.data.person.Person;
+
+/**
+ * Represents the addition or removal of a tag from a person 
+ * Guarantees: immutable
+ */
+public class Tagging {
+    public enum Action {
+        ADDING, REMOVING
+    }
+    
+    private Person person; 
+    private Tag tag; 
+    private Action action; 
+    
+    public Tagging (Person person, Tag tag, Action action) {
+        this.person = person;
+        this.tag = tag;
+        this.action = action;
+    }
+    
+    public Person getPerson() {
+        return person;
+    }
+    
+    public Tag getTag() {
+        return tag; 
+    }
+    
+    public Action getAction() {
+        return action; 
+    }
+    
+    /**
+     * Retrieve the shorthand representation of the provided action 
+     * @param action
+     * @return 
+     */
+    public static String getActionShorthand(Action action) {
+        switch (action) {
+        case ADDING:
+            return "+";
+        case REMOVING:
+        default:
+            return "-";
```

Comment:
This is a bit `too clever` :-p 
This is better (as adding and removing are similar operations, they should be handled similarly):
```java
case REMOVING:
    return "-";
default:
    throw new RuntimeException("Unknown tag operation"); 
```